### PR TITLE
[zedsdk]Add new port.

### PR DESCRIPTION
**Pull request by JackBoosY**
Needs to install zed sdk manually in Windows first.

Related: #8702.

### CODE REVIEW

File: ports/zedsdk/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        FILENAME "${ZED_SDK_PACKAGE_NAME}"
+        SHA512 ${ZED_SDK_SHA512}
+    )
+    
+    if (VCPKG_TARGET_IS_WINDOWS)
+        # ZED-SDK windows package does not support quite/auto install
+        message(FATAL_ERROR "Please execute ${DOWNLOADS}/${ZED_SDK_PACKAGE_NAME} manually and restart computer before install port zedsdk.")
+    else()
+        # Modify the package permission
+        vcpkg_execute_required_process(
```
Comment (JackBoosY) [inline ports/zedsdk/portfile.cmake]: It’s not appropriate to use the elevated permission command here, but I don’t know how to change it.

File: ports/zedsdk/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+find_program(ZEDFU
+    NAMES ZEDfu.exe ZEDfu
+    PATHS
+      $ENV{ZED_SDK_ROOT_DIR}
+      /usr/local/zed
+    PATH_SUFFIXES app/ZEDfu tools
+    NO_DEFAULT_PATH
+)
+
+message("ZED_SDK_ROOT_DIR: $ENV{ZED_SDK_ROOT_DIR}")
```
Comment (vicroms) [inline ports/zedsdk/portfile.cmake]: I believe this port should use a `environment-overrides.cmake` in order for the `ZED_SDK_ROOT_DIR` environment variable to be available when building. 

@cbezault is this correct? 

For reference:
https://github.com/microsoft/vcpkg/blob/master/ports/ogre-next/environment-overrides.cmake

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/zedsdk/portfile.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/zedsdk/portfile.cmake]: Any conclusion?

Comment (vicroms) [reply ports/zedsdk/portfile.cmake]: All environment variables are cleared when building ports on Windows, right now we haven't found a way to do the same in Linux/MacOS but we are planning on fixing that at some point.

For that reason, environment variables that need to be kept for a specific port should be declared in an `environment-overrides.cmake` file.

**Pull request closed without merging**